### ISSUE TITLE
Console macro should accept formatting strings

### ISSUE THREAD

**Issue by Diggsey** [koute/stdweb#130]
`console!` is a macro for the sole purpose of accepting variable numbers of arguments. We should instead follow the convention laid down by the other related macros (`print`, `println`, `eprint`, `eprintln`, `write`, `writeln`) and accept a formatting string.

**Comment by Pauan** [comment]
No, `console` isn't a `println` replacement (if you want that then *just use `println`*).

The purpose of `console` is to *pass through its arguments unchanged*, so that you can get a better debugging experience: browsers format things differently when you pass in an object/array/etc. (it lets you click on a small arrow to expand the properties, lets you view the prototype, etc.)

This is also important for debugging the various `Reference` types, because right now if you use `println` they get output as `Reference(42)` (the number varies), which is useless. But if you use `console` you get to see the actual JavaScript object.

**Comment by koute** [comment]
Just an @Pauan said - the `console!` macro isn't meant to be a general-purpose logging facility.

What we should have, though, is (optional) integration with the `log` crate, so you'd be able to do `info!(...)` and have that land it the dev console.

**Comment by Pauan** [comment]
@koute Yeah, there was already some work done on that, we can probably just reuse their implementation.

Also, I think we should change the docs for `console` to make it more clear what its exact purpose is, since as @Diggsey pointed out, it's easy to confuse it with the other printing macros.
